src/partition_mesh.py

```diff
@@ -83,387 +45,423 @@ def __str__(self):
-            (a**2 + b**2 - c**2 - d**2, 2 * (b * c - a * d), 2 * (b * d + a * c)),
-            (2 * (b * c + a * d), a**2 + c**2 - b**2 - d**2, 2 * (c * d - a * b)),
-            (2 * (b * d - a * c), 2 * (c * d + a * b), a**2 + d**2 - b**2 - c**2)))
-        for i, x in enumerate(mesh):  # Translate & Rotate
-            x -= pA
-            x = x @ rotMat
-            mesh[i] = x
-        return mesh[:, :-1]
-
-
-def partition_metis(mesh: Mesh, numparts: int):
-    """
-    Partitions a mesh using METIS. This does not call METIS directly,
-    but instead uses a small C++ Wrapper libmetisAPI.so for convenience.
-    This shared library must be provided if this function should be called.
-    """
-    cellPtr = [0]
-    cellData = []
-    if len(mesh.cells) == 0:
-        logging.warning("No topology information provided. Partitioning with metis will likely provide bad partition")
-    for i in range(len(mesh.cells)):
-        cell = mesh.cells[i]
-        cellData += list(cell)
-        cellPtr.append(cellPtr[-1] + len(cell))
-    binpath = os.path.dirname(__file__)
-    libpath = os.path.normpath(os.path.join(binpath, "../lib"))
-    os_type = platform.system()
-    if os_type == "Linux":
-        ext = ".so"
-    elif os_type == "Darwin":  # MacOS
-        ext = ".dylib"
-    else:
-        raise Exception("Unknown OS type")
-    if os.path.isfile(os.path.join(binpath, "libmetisAPI"+ext)):
-        libmetispath = os.path.join(binpath, "libmetisAPI"+ext)
-    elif os.path.isfile(os.path.join(libpath, "libmetisAPI"+ext)):
-        libmetispath = os.path.join(libpath, "libmetisAPI"+ext)
-    else:
-        raise Exception("libmetisAPI"+ext+" cannot found!")
-    libmetis = cdll.LoadLibrary(libmetispath)
-    idx_t = c_int if libmetis.typewidth() == 32 else c_longlong
-    cell_count = idx_t(len(mesh.cells))
-    point_count = idx_t(len(mesh.points))
-    num_parts = idx_t(numparts)
-    partition = (idx_t * len(mesh.points))()
-    cell_ptr = (idx_t * len(cellPtr))(*cellPtr)
-    cell_data = (idx_t * len(cellData))(*cellData)
-    libmetis.partitionMetis(cell_count, point_count, cell_ptr, cell_data, num_parts, partition)
-    return np.ctypeslib.as_array(partition)
-
-
-def partition_uniform(mesh: Mesh, numparts: int):
-    """
-    Partitions a mesh assuming it is uniform. It must be two-dimensional
-    but is allowed to be layed out anyhow in three dimensions.
-    """
-    mesh = mesh.points[:]
-    mesh = reduce_dimension(np.array(mesh))
-    if len(mesh[0]) == 3:
-        logging.warning("Mesh is not uniform. Falling back to meshfree method")
-        return None
-    min_point = np.amin(mesh, 0)
-    max_point = np.amax(mesh, 0)
-    big_dim = 0 if max_point[0] - min_point[0] >= max_point[1] - min_point[1] else 1
-    small_dim = 1 - big_dim
-
-    def prime_factors(n):
-        """ Straight from SO"""
-        i = 2
-        factors = []
-        while i * i <= n:
-            if n % i:
-                i += 1
+class MeshPartitioner:
```

Comment:
Same as above: A class description would be nice.

Reply:
Class description added